airbyte-cdk/python/unit_tests/sources/declarative/test_factory.py

```diff
@@ -559,5 +572,14 @@ def test_add_fields(self):
         config = parser.parse(content)
         component = factory.create_component(config["the_stream"], input_config)()
         assert isinstance(component, DeclarativeStream)
-        expected = [AddFields([AddedFieldDefinition(["field1"], "static_value")])]
-        assert expected == component._transformations
+        expected = [
+            AddFields(
+                fields=[
+                    AddedFieldDefinition(
+                        path=["field1"], value=InterpolatedString(string="static_value", default="static_value", options={}), options={}
```

Comment:
An interesting side effect of switching to dataclasses is that since `options` is passed everywhere, this used to be a string type, but now with options being passed, our type inference switches this to an interpolated string. Functionally, nothing should change, so this isn't a big issue, but worth pointing out

Reply:
this is better since it'll allow us to interpolate on the value right?

Reply:
it is yeah. not sure why it was working before coming in as a string, but yeah this would be a net positive.